packages/client/src/containers/RelatedItems/RelatedItems.Container.js

```diff
@@ -1,33 +1,74 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
 import PropTypes from 'prop-types';
 import ProductCard from '../../components/ProductCard/ProductCard.component';
 import { apiURL } from '../../apiURL';
+import { ViewBackCarousel } from '../../components/ViewCarousel/ViewBackCarousel/ViewBackCarousel';
+import { ViewAfterCarousel } from '../../components/ViewCarousel/ViewForwadCarousel/ViewAfterCarousel';
 import './RelatedItems.styles.css';
 
 export const RelatedItems = ({ category }) => {
   const [categoryProducts, setCategoryProducts] = useState([]);
+  const [offSet, setOffset] = useState(0);
+  const [dataFlag, setDataFlag] = useState(true);
 
-  useEffect(() => {
-    const fetchProductsByCategory = async () => {
-      const response = await fetch(`${apiURL()}/products?category=${category}`);
-      const categoryProductData = await response.json();
+  const fetchProductsByCategory = useCallback(async () => {
+    const response = await fetch(
+      `${apiURL()}/products?category=${category}&limit=3&offset=${offSet}`,
+    );
+    const categoryProductData = await response.json();
+
+    if (categoryProductData.data.length < 3) {
+      fetchAllProducts();
+    } else {
+      setDataFlag(true);
+      setCategoryProducts(categoryProductData.data);
```

Comment:
What if there is exactly 3 products? It will set the flag to true, but when you click next you'll get nothing back. 

Instead, it would be great if your API actually returned a count as well. So the response could look like:
```
{
  status: 200,
  total: 13,
  data: [
    product1,
    product2,
    product3
  ].
}
```

That way you don't need the flag, you can simply check if `total` is larger than `offset` .

Reply:
If there are 3 it will show three items and when we will click the offset will increase there will be no related data then it will display the first three products from all products and disables the forward button

Reply:
As a user, I don't think I would understand that those next 3 are **not** related items. I think it would be much better if we had a way to count how many related items we have and act on that instead of some fallback behaviour.